### PR TITLE
Remove publish step from circleCI, will publish from release repo

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Where is the release repo?

**Comment by bogdandrutu** [comment]
@codeboten https://github.com/open-telemetry/opentelemetry-collector-releases

File: .circleci/config.yml [hunk 1]
```diff
@@ -75,40 +52,6 @@ commands:
...
-      - run:
-          name: Login to Docker Hub
-          command: docker login -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD
-      - run:
-          name: Push image
-          command: |
-            docker push otel/<< parameters.repo >>:<< parameters.tag >>
-            docker push otel/<< parameters.repo >>:latest
-
-  github_issue_generator:
```
Comment (codeboten) [inline .circleci/config.yml:101]: This step appears unrelated to publishing releases.

Comment (bogdandrutu) [reply .circleci/config.yml:101]: Nobody uses it :D was not a step but a command so if nobody is calling it it does not get executed

**Review by dmitryax — APPROVED** [review]
